main.go

```diff
@@ -94,6 +94,15 @@ func main() {
 	// Pages
 	pages.AddPage("Main", flex, true, true)
 
+	// Start our background refresh timer
+	go func() {
+		for {
+			text.Clear()
+			text.SetText(getPromText(ctx))
+			time.Sleep(time.Second * 5)
+		}
+	}()
```

Comment:
You may want to use `time.AfterFunc` here instead. It gets rid of the infinite goroutine and allows you to cancel/modify the timer.